internal/service/alarms/api/openapi.yaml

```diff
@@ -0,0 +1,756 @@
+openapi: 3.0.3
+info:
+  title: O2IMS Infrastructure Monitoring Alarms API
+  version: 1.0.0
+  description: API for O2IMS Infrastructure Monitoring Alarms
+  contact:
+    name: Red Hat
+    url: https://www.redhat.com
+
+externalDocs:
+  description: O-RAN apis from O-RAN.WG6.O2IMS-INTERFACE-R003-v06.00 (June 2024)
+  url: https://specifications.o-ran.org/download?id=674
+
+servers:
+  - url: http://localhost:8080
+    description: O-RAN Alarms Server
+
+tags:
+  - name: alarms
+    description: Alarm management
+  - name: subscriptions
+    description: Alarm subscription management
+  - name: probableCauses
+    description: Probable cause information
+  - name: internal
+    description: |
+      Internal endpoints. This is tag is also useful generating client code.
+
+paths:
+  /O2ims_infrastructureMonitoring/v1/alarms:
+    get:
+      operationId: GetAlarms
+      summary: Retrieve the list of alarms
+      tags:
+        - alarms
+      parameters:
+        - in: query
+          name: filter
+          schema:
+            type: string
+          allowReserved: true
+          description: Optional filter parameter. Only the attributes in AlarmEventRecord are filterable.
+          example: (eq,perceivedSeverity,0)
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/AlarmEventRecord'
+        '400':
+          description: Bad request
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '403':
+          description: Forbidden.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '404':
```

Comment:
I've seen this 404 for lists when filtering is used and there no matches. But we can also return an empty list with 200 OK for this case (with filtering). If so, we won't need this response code.

Both are fine in my opinion but given that the specification explicitly calls for 200 OK for empty lists. I think we can go for the second approach. (200 OK empty list with and without filtering).

Unless your intention with this response code was to question the existence of the endpoint. Hmm, I'd say 500 is better in that case.

Reply:
> I've seen this 404 for lists when filtering is used and there no matches. But we can also return an empty list with 200 OK for this case (with filtering)

Good point! Basically if no match found do we simply return an empty list instead of 404?  Will follow spec and remove 404.